### DISCUSSION TITLE
How to utilize laravel-mix with nonce feature

### DISCUSSION THREAD

**Discussion by sys-auditing** [spatie/laravel-csp#96] [Q&A]
Hi, 

how to use ```laravel-mix``` with nonce feature?

Thank you

**Comment by rich06** [comment]
Did you resolve this? I have the same issue with scripts compiled by Laravel Mix dynamically creating styles - how to apply a nonce to these items?

**Reply by sys-auditing** [reply]
Hi @rich06 , 
i have just add
 ```<script src="{{ asset('js/app.js') }}" defer nonce="{{ csp_nonce() }}"></script>```
 ```<link href="{{ asset('css/app.css') }}" rel="stylesheet" nonce="{{ csp_nonce() }}"> ```

**Comment by rich06** [comment]
Hi @sys-auditing 
Thanks for your message and yes I have this in the blade:-
 
```
<link nonce="{{ csp_nonce() }}" href="{{ mix('css/app.css') }}" rel="stylesheet">    
<script nonce="{{ csp_nonce() }}" src="{{ mix('js/app.js') }}"></script>

```
But loading the page generates a lot of these errors in the console:-

`Content Security Policy: The page's settings blocked the loading of a resource at inline ("style-src").
`
The errors seem to come from modules dynamically adding style tags.. did you not get this problem?

**Reply by sys-auditing** [reply]
yes, i had.
But i removed all inline styles, inline actions ...

**Reply by rich06** [reply]
OK thanks for the info

**Reply by sys-auditing** [reply]
You are welcome, i hope the info helped you to resolve the issue :).
Instead you can looking for ```style-src``` that catch the issue.
